libs/postgres_connection/src/lib.rs

```diff
@@ -0,0 +1,222 @@
+use anyhow::Context;
+use std::fmt;
+use url::Host;
+
+pub fn parse_host_port<S: AsRef<str>>(host_port: S) -> Result<(Host, Option<u16>), anyhow::Error> {
+    let url = url::Url::parse(&format!("postgres://{}", host_port.as_ref()))
+        .context("Invalid host:port pair specified")?;
+    let host = url.host().context("Host is not specified")?.to_owned();
+    Ok((host, url.port()))
```

Comment:
Good idea, I forgot about splitting from the right. Splitting from the left does not work with `[::1]`, so I discarded it altogether.

Reply:
Unfolding since I still have something to add here.

```rust
>> "-0".parse::<u16>() // can't reproduce
Err(ParseIntError { kind: InvalidDigit })
>> "+10".parse::<u16>() // this is unfortunate, yes
Ok(10)
```

I think we could add an `is_numeric` check on top if you'd like to.

Reply:
Good catch, thank you. I've missed that `if` while reading `from_str_radix`.